src/main/java/io/servertap/api/v1/ServerApi.java

```diff
@@ -206,6 +204,116 @@ public static void saveWorld(Context ctx) {
+            method = HttpMethod.GET,
+            tags = {"Server"},
+            headers = {
+                    @OpenApiParam(name = "key")
+            },
+            pathParams = {
+                    @OpenApiParam(name = "uuid", description = "The UUID of the World to download")
+            },
+            responses = {
+                    @OpenApiResponse(status = "200", content = @OpenApiContent(type = "application/zip")),
+            }
+    )
+    public static void downloadWorld(Context ctx) throws IOException {
+        org.bukkit.Server bukkitServer = Bukkit.getServer();
+
+        UUID worldUUID = ValidationUtils.safeUUID(ctx.pathParam("uuid"));
+        if (worldUUID == null) {
+            throw new BadRequestResponse(Constants.INVALID_UUID);
+        }
+
+        org.bukkit.World world = bukkitServer.getWorld(worldUUID);
+
+        if (world != null) {
+            File folder = world.getWorldFolder();
+
+            // Set headers for proper zip download
+            ctx.header("Content-Disposition", "attachment; filename=\"" + folder.getName() + ".zip\"");
+            ctx.header("Content-Type", "application/zip");
+
+
+            ZipOutputStream zip = new ZipOutputStream(ctx.res.getOutputStream());
+            addFolderToZip(folder, zip, folder.getAbsolutePath(), null);
+            zip.close();
+        }
+    }
+
+    @OpenApi(
+            path = "/v1/worlds/download",
+            summary = "Downloads a Zip compressed archive of all the worlds' folders",
+            method = HttpMethod.GET,
+            tags = {"Server"},
+            headers = {
+                    @OpenApiParam(name = "key")
+            },
+            responses = {
+                    @OpenApiResponse(status = "200", content = @OpenApiContent(type = "application/zip")),
+            }
+    )
+    public static void downloadWorlds(Context ctx) throws IOException {
+        org.bukkit.Server bukkitServer = Bukkit.getServer();
+
+        Plugin pluginInstance = bukkitServer.getPluginManager().getPlugin("ServerTap");
+
+        ctx.header("Content-Disposition", "attachment; filename=\"worlds.zip\"");
+        ctx.header("Content-Type", "application/zip");
+
+        ZipOutputStream zip = new ZipOutputStream(ctx.res.getOutputStream());
+
+        for (org.bukkit.World world : Bukkit.getWorlds()) {
+            try {
+
+                File folder = world.getWorldFolder();
+
+                addFolderToZip(folder, zip, folder.getAbsolutePath(), null);
+
+            } catch (Exception e) {
+                // Just warn about the issue
+                log.warning(String.format("Couldn't save World %s %s", world.getName(), e.getMessage()));
+                throw new InternalServerErrorResponse("Couldn't download world " + world.getName() + ": " + e.getMessage());
+            }
+        }
+        try {
+            zip.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Useless to throw a

Reply:
I would probably do a `log.severe(e)` here instead of `e.printStackTrace()`

Reply:
Removed the try-catch, kinda useless.